dexmaker/src/main/java/com/android/dx/stock/ProxyBuilder.java

```diff
@@ -813,8 +811,9 @@ private void getMethodsToProxy(Set<MethodSetEntry> sink, Set<MethodSetEntry> see
         }
     }
 
-    private static <T> String getMethodNameForProxyOf(Class<T> clazz) {
-        return clazz.getName().replace(".", "/") + "_Proxy";
+    private static <T> String getMethodNameForProxyOf(Class<T> clazz, Set<Class<?>> interfaces) {
+        String interfacesHash = Integer.toHexString(interfaces.hashCode());
+        return clazz.getName().replace(".", "/") + "_" + interfacesHash + "_Proxy";
```

Comment:
Why is this necessary? You already added the interfaces to the proxy-key

Reply:
In the original implementation, since the proxy classes based on the same class would have the same name, the code below would always return the first generated class.

https://github.com/linkedin/dexmaker/blob/ab6d870302154e502a693337944d550a31cbb2f6/dexmaker/src/main/java/com/android/dx/stock/ProxyBuilder.java#L335

Reply:
makes sense